src/symbolize/gimli.rs

```diff
@@ -372,28 +372,31 @@ cfg_if::cfg_if! {
         fn native_libraries() -> Vec<Library> {
             let mut ret = Vec::new();
             unsafe {
-                libc::dl_iterate_phdr(Some(callback), &mut ret as *mut _ as *mut _);
+                libc::dl_iterate_phdr(Some(callback), &mut ret as *mut Vec<_> as *mut _);
             }
             return ret;
         }
 
+        // SAFETY: `info` and `vec` should be valid pointers
         unsafe extern "C" fn callback(
             info: *mut libc::dl_phdr_info,
             _size: libc::size_t,
             vec: *mut libc::c_void,
         ) -> libc::c_int {
+            let info = &*info;
             let libs = &mut *(vec as *mut Vec<Library>);
-            let name = if (*info).dlpi_name.is_null() || *(*info).dlpi_name == 0{
+            let is_main_prog = info.dlpi_name.is_null() || info.dlpi_name.read() == 0;
```

Comment:
I would prefer to use `*` here instead of a method to read the pointer.

Reply:
Will do although I personally prefer postfix deref operator, but we can't change it now.